packages/wallets/streambird/src/adapter.ts

```diff
@@ -0,0 +1,189 @@
+
+  private _params: StreambirdParams;
+  private _connecting: boolean;
+  private _wallet: Streambird | null;
+  private _publicKey: PublicKey | null;
+  private _readyState: WalletReadyState = typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
+  
+  constructor({ params = {} }: StreambirdWalletAdapterConfig = {}) {
+    super();
+    this._connecting = false;
+    this._publicKey = null;
+    this._wallet = null;
+    this._params = params;
+  }
+
+  get publicKey(): PublicKey | null {
+    return this._publicKey;
+  }
+
+  get connecting(): boolean {
+    return this._connecting;
+  }
+
+  get connected(): boolean {
+    return !!this._wallet?.isLoggedIn;
+  }
+
+  get readyState(): WalletReadyState {
+    return this._readyState;
+  }
+
+  async connect(): Promise<void> {
+    try {
+      if (this.connected || this.connecting) return;
+      if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+      this._connecting = true;
+
+      let StreambirdClass: typeof Streambird;
+
+      try {
+        ({ default: StreambirdClass } = await import('@streambird/solana-embed-sdk'));
+      } catch (error: any) {
+        throw new WalletLoadError(error?.message, error);
+      }
+
+      let wallet: Streambird;
+
+      try {
+        wallet = window.streambird || new StreambirdClass();
+      } catch (error: any) {
+        throw new WalletConfigError(error?.message, error);
+      }
+
+      if (!wallet.isInitialized) {
+        try {
+          await wallet.init(this._params);
+        } catch (error: any) {
+          throw new WalletConnectionError(error?.message, error);
+        }
+      }
+
+      let base58PublicKey: string = '';
+
+      try {
+        base58PublicKey = await wallet.login();
+      } catch (error: any) {
+        throw new WalletConnectionError(error?.message, error);
+      }
+
+      let existingPublicKey: PublicKey | null = null;
+
+      try {
+        existingPublicKey = new PublicKey(base58PublicKey);
+      } catch (error: any) {
+        throw new WalletPublicKeyError(error?.message, error);
+      }
+
+      this._wallet = wallet;
+      this._publicKey = existingPublicKey;
+      
+      this.emit('connect', existingPublicKey);
+    } catch (error: any) {
+        this.emit('error', error);
+        throw error;
+    } finally {
+        this._connecting = false;
+    }
+  }
+
+  async disconnect(): Promise<void> {
+    const wallet = this._wallet;
+
+    if (wallet) {
+      this._wallet = null;
+      this._publicKey = null;
+
+      try {
+        if(wallet.isLoggedIn) {
+          await wallet.cleanUp();
+        }
```

Comment:
If it's not actually required for the adapter to wait until `cleanUp` finishes, it might be good to just ignore the promise. What would actually lead to an error here?